src/NLSModels.jl

```diff
@@ -21,6 +21,7 @@ type NLSCounters
 end
 
 nls_counters(nls :: AbstractNLSModel) = nls.counters
+counters(nls :: AbstractNLSModel) = nls_counters(nls).counters
```

Comment:
Do we want the counters to be aliased this way? Won't it make it difficult to compare, e.g., the number of objective evaluations when running a solver on a set of problems?

Reply:
It's this way because of models like `SlackModel`, which uses the counters of the submodel. In particular I'm using it to increment the counters. One alternative would be create something like
`increment_counter(c :: Symbol)`, and remove all `counters(nlp)` and `nls_counters(nls)`.

Reply:
What I mean is: do `NLSModel` instances also have `neval_obj`, etc., counters?

Reply:
[Yes](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/abfb806e1d1af903f2b2bfe5bdcd2ccb810478f0/src/NLSModels.jl#L43). We probably need to test whether all this referencing is being done right.